perses/rjmc/geometry.py

```diff
@@ -291,8 +291,9 @@ def _logp_propose(self, top_proposal, old_positions, beta, new_positions=None, d
                 logZ_r = np.log((np.sqrt(2*np.pi)*(sigma_r/units.angstroms))) # CHECK DOMAIN AND UNITS
                 logp_r = self._bond_logq(r, bond, beta) - logZ_r
             else:
-                constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
-                r = constraint #set bond length to exactly constraint
+                if direction == 'forward':
+                    constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
```

Comment:
Well, that code is only reachable if there is a bond but no bond information (which is how `ParmEd` deals with bond constraints). If there's a bond, but neither parameters nor a constraint, the system is not going to work.

`r` is the bond length, which can either come from sampling a distribution defined by the bond parameters, or the constraint. But either way, it's necessary for the rest.